### PR TITLE
[quant] Make version 1 the default for get_default_qat_qconfig

### PR DESCRIPTION
**Pull request by supriyar**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #63043

Summary:
In version 1 we use the fused module/operator during QAT. Making this the default for all QAT runs going forward.

Older models saved after prepare_qat_fx can still load their state_dict into a model prepared using version 1.
The state_dict will still have the same attribute for the observer/fake_quant modules.

There may be some numerics difference between the old observer code in observer.py and the new fused module that was
re-written in C++/CUDA to perform observe + fake_quantize.

This PR also updates the test to check for the new module instead of the default FakeQuantize module.
Note: there are also some changes to make the operator work for multi-dim per-channel quantization + updated the test for that.

Test Plan:
python test/test_quantization.py TestSerialization.test_default_qat_qconfig

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30232222](https://our.internmc.facebook.com/intern/diff/D30232222)

### CODE REVIEW

**Comment by supriyar** [comment]
@supriyar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30232222).

**Comment by supriyar** [comment]
@supriyar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30232222).

**Review by raghuramank100 — APPROVED** [review]

**Comment by supriyar** [comment]
@supriyar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30232222).

**Comment by supriyar** [comment]
@supriyar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30232222).
